src/Mantasflowers.Services/Services/Order/OrderService.cs

```diff
@@ -0,0 +1,67 @@
+﻿using AutoMapper;
+using Mantasflowers.Contracts.Order.Request;
+using Mantasflowers.Contracts.Order.Response;
+using Mantasflowers.Services.Repositories;
+using Mantasflowers.Services.Services.Exceptions;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Storage;
+using System;
+using System.Threading.Tasks;
+
+namespace Mantasflowers.Services.Services.Order
+{
+    public class OrderService : IOrderService
+    {
+        private readonly IOrderRepository _orderRepository;
+        private readonly IMapper _mapper;
+
+        public OrderService(IOrderRepository orderRepository, IMapper mapper)
+        {
+            _orderRepository = orderRepository;
+            _mapper = mapper;
+        }
+
+        public async Task<Domain.Entities.Order> CreateOrderAsync(PostCreateOrderRequest request)
+        {
+            var order = _mapper.Map<Domain.Entities.Order>(request);
+            order.TemporaryPasswordHash = "stoppolice";
+
+            try
+            {
+                order = await _orderRepository.CreateAsync(order);
+            }
+            catch (DbUpdateException)
+            {
+                throw new FailedToAddDatabaseResourceException("Failed to create order");
```

Comment:
the `CreateAsync` will throw the exception it self. Whats the point of this exception? To have a custom message?

Reply:
For consistency, followed as was written in [UserService.cs](https://github.com/mantasflowers/Mantasflowers.Backend/blob/payment-shopping_cart-setup/src/Mantasflowers.Services/Services/User/UserService.cs).
```cs
// TODO: [NFR] transactions? Exception catching? (Uid could exist or concurrent add user requests)
public async Task<PostCreateUserResponse> CreateUserAsync(string uid)
{
    var user = new Domain.Entities.User
    {
        Uid = uid
    };

    try
    {
        await _userRepository.CreateAsync(user);
    }
    catch (DbUpdateException)
    {
        throw new FailedToAddDatabaseResourceException("Failed to create user");
    }

    var resposne = _mapper.Map<PostCreateUserResponse>(user);

    return resposne;
}
```